symfony/plugins/orangehrmAdminPlugin/modules/admin/actions/deleteLocationsAction.class.php

```diff
@@ -27,24 +27,26 @@ public function getLocationService() {
 		return $this->locationService;
 	}
 
-	public function execute($request) {
-                $form = new DefaultListForm();
-                $form->bind($request->getParameter($form->getName()));
-		$toBeDeletedLocationIds = $request->getParameter('chkSelectRow');
-
-		if (!empty($toBeDeletedLocationIds)) {
-
-			foreach ($toBeDeletedLocationIds as $toBeDeletedLocationId) {
-                            if ($form->isValid()) {
-				$location = $this->getLocationService()->getLocationById($toBeDeletedLocationId);
-				$location->delete();
-                                $this->getUser()->setFlash('success', __(TopLevelMessages::DELETE_SUCCESS));
-                            }
-			}
-		}
-
-		$this->redirect('admin/viewLocations');
-	}
+    public function execute($request) {
+        $form = new DefaultListForm();
+        $form->bind($request->getParameter($form->getName()));
+        $toBeDeletedLocationIds = $request->getParameter('chkSelectRow');
+
+        if (!empty($toBeDeletedLocationIds)) {
+
+            foreach ($toBeDeletedLocationIds as $toBeDeletedLocationId) {
+                if ($form->isValid()) {
+                    $location = $this->getLocationService()->getLocationById($toBeDeletedLocationId);
+                    if ($location instanceof Location) {
+                        $location->delete();
+                    }
+                    $this->getUser()->setFlash('success', __(TopLevelMessages::DELETE_SUCCESS));
```

Comment:
Flash in foreach loop?
flash looks like it will be set 20 times if 20 Ids are passed?